Components/Sensors/Inc/BarometerTask.hpp

```diff
@@ -0,0 +1,55 @@
+/**
+ ******************************************************************************
+ * File Name          : BarometerTask.hpp
+ * Description        :
+ ******************************************************************************
+*/
+#ifndef SOAR_SENSOR_BAROMETER_TASK_HPP_
+#define SOAR_SENSOR_BAROMETER_TASK_HPP_
+/* Includes ------------------------------------------------------------------*/
+#include "Task.hpp"
+#include "Data.h"
+#include "SystemDefines.hpp"
+
+
+/* Macros/Enums ------------------------------------------------------------*/
+enum BARO_TASK_COMMANDS {
+	BARO_NONE = 0,
+	BARO_REQUEST_NEW_SAMPLE,// Get a new barometer sample, task will be blocked for polling time
+	BARO_REQUEST_TRANSMIT,	// Send the current barometer data over the Radio
+	BARO_REQUEST_DEBUG		// Send the current barometer data over the Debug UART
+};
+
+/* Class ------------------------------------------------------------------*/
+class BarometerTask : public Task
+{
+public:
+	static BarometerTask& Inst() {
+		static BarometerTask inst;
+		return inst;
+	}
+
+	void InitTask();
+
+protected:
+	static void RunTask(void* pvParams) { BarometerTask::Inst().Run(pvParams); } // Static Task Interface, passes control to the instance Run();
+
+	void Run(void* pvParams);	// Main run code
+	
+	void HandleCommand(Command& cm);
+	void HandleRequestCommand(uint16_t taskCommand);
+
+	// Sampling
+	void SampleBarometer();
+	uint16_t ReadCalibrationCoefficients(uint8_t PROM_READ_CMD);
+
+	// Data
+	BarometerData* data;
```

Comment:
why keep a pointer to the struct and malloc required memory? why not have it stored locally, statically allocated, and pass references/pointers when required?

Reply:
We're using malloc but never freeing this data, to avoid using up the stack in this case.

As it is never freed this should have minimal performance impact.